src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -57,110 +38,110 @@ public class ParseAndCompareBenchmarkResults {
+     * @return the composed regression detection summary.
      */
-    public static Map<String, BigDecimal> getThresholdMap(String benchmarkResultPrevious, String benchmarkResultNew) throws Exception {
-        Map<String, BigDecimal> thresholdMap = new HashMap<>();
-        for (String keyWord : BENCHMARK_SCORE_KEYWORDS) {
-            IonList rawDataPrevious = (IonList) parseScore(benchmarkResultPrevious, keyWord).get(RAW_DATA);
-            IonList rawDataNew = (IonList) parseScore(benchmarkResultNew, keyWord).get(RAW_DATA);
-            BigDecimal thresholdPrevious = getThresholdScore((IonList) rawDataPrevious.get(0));
-            BigDecimal thresholdNew = getThresholdScore((IonList) rawDataNew.get(0));
-            if (thresholdPrevious.compareTo(thresholdNew) < 0) {
-                thresholdMap.put(keyWord, thresholdPrevious);
-            } else {
-                thresholdMap.put(keyWord, thresholdNew);
-            }
+    private static String comparisonSummary(Map<String, Double> result) {
+        StringBuilder summary = new StringBuilder();
+        for (Map.Entry<String, Double> comparisonResult : result.entrySet()) {
+            summary.append(String.format("There is %.2f%% regression on %s.\n", comparisonResult.getValue() * 100, comparisonResult.getKey()));
         }
-        return thresholdMap;
+        return summary.toString();
     }
 
     /**
-     * Get threshold score by applying (minScore - maxScore)/maxScore to a list of raw data in benchmark result.
-     *
-     * @param rawDataList is an Ion List contains performance scores from multiple iterations of benchmark process.
-     * @return calculated threshold score.
+     * This method uses two-sample T-test method to detect performance regression.
+     * @param before represents the benchmark results from benchmarking the previous ion-java.
+     * @param after represents the benchmark results from benchmarking the ion-java with the new changes.
+     * @return regression detection result. If the regression is detected, then the regression value will be returned, else 0 will be returned.
      */
-    private static BigDecimal getThresholdScore(IonList rawDataList) {
-        List<BigDecimal> rawData = new ArrayList<>();
-        for (int i = 0; i < rawDataList.size(); i++) {
-            IonDecimal score = (IonDecimal) rawDataList.get(i);
-            rawData.add(score.bigDecimalValue());
+    public static double detectRegression(double[] before, double[] after) {
+        TTest tTest = new TTest();
+        // Perform two-sample t-test
+        double pValue = tTest.tTest(before, after);
+        // Calculate means of both datasets
+        double meanBefore = StatUtils.mean(before);
+        double meanAfter = StatUtils.mean(after);
+        // Calculate the difference in means (regression value)
+        double regressionValue = (meanAfter - meanBefore) / meanBefore;
```

Comment:
The code would be more logical/readable if these lines were moved into the `if (pValue <0.05) {` block.